src/main/java/org/folio/circulation/infrastructure/storage/CirculationPolicyRepository.java

```diff
@@ -100,6 +99,26 @@ protected CirculationPolicyRepository(CollectionResourceClient policyStorageClie
         "Unable to apply circulation rules for unknown item"));
     }
 
+    if (user.getPatronGroupId() == null) {
+      log.error("PatronGroupId is null for user {}", user.getId());
+      return completedFuture(failedDueToServerError("Unable to apply circulation rules with user that is patronGroup is null"));
+    }
+
+    if (item.getLocationId() == null) {
+      log.error("LocationId is null for item {}", item.getItemId());
+      return completedFuture(failedDueToServerError("Unable to apply circulation rules with item that is locationId is null"));
+    }
+
+    if (item.determineLoanTypeForItem() == null) {
+      log.error("LoanTypeId is null for item {}", item.getItemId());
+      return completedFuture(failedDueToServerError("Unable to apply circulation rules with item that is loanTypeId is null"));
```

Comment:
Unable to apply circulation rules to an item which loan type can not be determined